Wikipedia/Code/WMFBaseExploreSectionController.m

```diff
@@ -204,61 +216,39 @@ - (AnyPromise*)fetchDataIgnoreError:(BOOL)ignoreError ignoreCurrentItems:(BOOL)i
             }
             return error;
         });
+        // NOTE: set items to placeholders only after promise has been set to indicate isFetching state
+        [self setItemsToPlaceholdersOrEmpty];
         return self.fetcherPromise;
     }
 }
 
 - (void)resetData {
-    self.fetchError = nil;
-    [self setItemsToPlaceholders];
+    [self setItemsToPlaceholdersOrEmpty];
 }
 
 #pragma mark - Utility
 
-- (void)setItemsToPlaceholders {
+- (void)setItemsToPlaceholdersOrEmpty {
     if (![self shouldShowPlaceholderCell]) {
+        self.items = @[];
```

Comment:
so, `items` is `nonnull`, and if the section doesn't support placeholders, it should be an empty array. I could either to do it explicitly here, or implicitly by falling through to the logic below and assuming the non-placeholder section has `numberOfPlaceholderCells == 0`.  I'll add a comment explaining the rationale